stellargraph/layer/attri2vec.py

```diff
@@ -182,13 +195,9 @@ def link_model(self):
 
         # Expose input and out sockets of the model, for target node:
         x_inp_dst = Input(shape=(1,))
-        output_embedding = Embedding(
-            self.input_node_num,
-            self.dims[self.n_layers],
-            input_length=1,
-            name="output_embedding",
-        )
-        x_out_dst = output_embedding(x_inp_dst)
+
+        assert isinstance(self._output_embedding, Embedding)
```

Comment:
Do we ever or is there ever a need for calling `link_model` or `node_model` directly rather than indirectly via a call to `build`? I understand this ticket is not exactly about this issue but it seems to me like the `*_model` methods should be "private". Anyway, just mentioning this, if you feel we need a ticket for this, otherwise, it all looks good as far as I can tell.

Reply:
Yeah good point, but I've filed https://github.com/stellargraph/stellargraph/issues/1116, because it seems more than just `Attri2Vec` have `node_model` and `link_model` methods in addition to `build` (also cc #1089): `APPNP`, `GAT`, `GCN`, `GraphSAGE`, `PPNP`.